[3.x] Improve validation of gamepads on Linux

JyKin opened the pull request:
EDIT: Changed idea and provide new fix after discussion

*Bugsquad edit:*
- Fixes #50273

old message:
cherry pick those commits from master :
8fe2ecf56581019e7f688200490637846d8bb3b7
b966ca6167beefc470c3469665200b542b1ea58a

madmiraal commented:
Note: #57612 (b966ca6167beefc470c3469665200b542b1ea58a) caused #59250. So, this probably shouldn't be merged without #59412 (bc17abceea50eafee49824b8d93fbc2a94facc26) which has not been approved, or some other better solution that more accurately identifies what is and what is not a gamepad.

JyKin committed:
Improve Linux Gamepad detection

Check if device has key: should filter out motion-control input from
controller like recent playstation or nintendo.

Check if device has direction input: joystick or dpad.

Check if device has no touch-like input: should filter out touchscreen
and fingerprint reader.

JyKin commented:
Ok, I went back and made a entirely new commit, without cherry-picking anything, with my solution.

akien-mga commented:
WDYT about this solution @madmiraal?

I'm a bit unsure about trying yet another approach which would differ from the one now used in 4.0 - but the code has diverged so it's possible that the 4.0 code isn't applicable here as is. The added booleans here are also nice for readability of what this is checking exactly.

madmiraal reviewed (changes requested):
This solution is problematic. It requires a device to have keys to be treated as a gamepad, which is the opposite of the current behaviour, which requires a device not to have keys to be treated as a gamepad.

More importantly, this PR should be made against master not 3.x. As identified in the OP, the code on master has diverged from 3.x, because #40591 and #57612 haven't been backported to 3.x. This is not a reason to start making changes to 3.x without first making them on master.

madmiraal commented:
Accurately detecting gamepads is hard. #40591 made our detection of gamepads the same as SDL's detection of gamepads. However, SDL's detection of gamepads (and devices in general) has since moved on:
https://github.com/libsdl-org/SDL/blob/9eb4d1f020bdc43e32375942120214eee5d70fee/src/core/linux/SDL_evdev_capabilities.c#L78-L105
It's also trying to copy the logic used by systemd:
https://github.com/systemd/systemd/blob/30338b8b66174a87727c8a8493ebf3a043f65b9d/src/udev/udev-builtin-input_id.c#L208-L295

Clearly, getting this right is going to take a more effort than just #40591, #57612, #59412 and #67414.

AThousandShips commented on platform/x11/joypad_linux.cpp:
```diff
@@ -330,16 +330,16 @@ void JoypadLinux::open_joypad(const char *p_path) {
-				!(test_bit(EV_ABS, evbit) &&
-						test_bit(ABS_X, absbit) && test_bit(ABS_Y, absbit) &&
-						test_bit(ABS_RX, absbit) && test_bit(ABS_RY, absbit))) {
+		// Check if the device supports basic gamepad events and not non-joystick events
```
```suggestion
		// Check if the device supports basic gamepad events and not non-joystick events.
```
Per the [comment style](https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html#comment-style-guide).
